secure.c

```diff
@@ -440,15 +441,25 @@ sec_out_mcs_data(STREAM s, uint32 selected_protocol)
 	out_uint32_le(s, g_keyboard_functionkeys);
 	out_uint8s(s, 64);	/* reserved? 4 + 12 doublewords */
 	out_uint16_le(s, 0xca01);	/* colour depth? */
-	out_uint16_le(s, 1);
+	out_uint16_le(s, 1);		/* product id */
 
-	out_uint32(s, 0);
-	out_uint8(s, g_server_depth);
-	out_uint16_le(s, 0x0700);
+	out_uint32(s, 0);		/* serial number */
+	out_uint8(s, g_server_depth);	/* this stuff is misaligned should be 2,2,2,64,1,1 */
+	out_uint16_le(s, 0x0700);	/* this strange constant is split across fields */
 	out_uint8(s, 0);
 	out_uint32_le(s, 1);
 	out_uint8s(s, 64);
 	out_uint32_le(s, selected_protocol);	/* End of client info */
+	if (g_dpi > 0)
+	{
+		/* Extended client info describing monitor geometry */
+		out_uint32_le(s, g_width * 254 / (g_dpi * 10)); /* desktop physical width */
+		out_uint32_le(s, g_height * 254 / (g_dpi * 10)); /* desktop physical height */
+		out_uint16_le(s, 0); /* orientation: landscape */
```

Comment:
LGTM but IMHO it would be better to add defines for desktop orientations according to https://msdn.microsoft.com/en-us/library/cc240510.aspx i.e. 
#define ORIENTATION_LANDSCAPE 0
#define ORIENTATION_PORTRAIT 90
etc.
What do you think?
P.S. It'd be a good PR policy to add proper defines from corresponding protocol documentation.